src/myq.js

```diff
@@ -3,287 +3,351 @@ const axios = require('axios');
+        .catch(({ response }) => ErrorHandler.prototype.parseBadResponse(response))
+    );
+  }
+
+  checkIsLoggedIn() {
+    return !!this.securityToken;
+  }
+
+  executeRequest(route, method, params, data) {
+    const isLoginRequest = route === constants.routes.login;
+    const headers = {
+      'Content-Type': 'application/json',
+      MyQApplicationId: constants.appId,
+    };
+
+    // If we aren't logged in or logging in, throw an error.
+    if (!isLoginRequest && !this.checkIsLoggedIn()) {
+      return ErrorHandler.prototype.returnError(13);
+    } else if (!isLoginRequest) {
+      // Add our security token to the headers.
+      headers.SecurityToken = this.securityToken;
+    }
+
+    let baseUrl = constants.deviceBase;
+
+    // Authentication routes need a different base URL.
+    if ([constants.routes.login, constants.routes.account].includes(route)) {
+      baseUrl = constants.authBase;
+    }
+
+    const config = {
+      method,
+      url: `${baseUrl}/${route}`,
+      headers,
+    };
+    if (data) {
+      config.data = data;
+    }
+    if (params) {
+      config.params = params;
+    }
+
+    return Promise.resolve(axios(config)).then(response => ({
+      returnCode: 0,
+      response,
+    }));
+  }
+
+  getAccountInfo() {
+    return this.executeRequest(constants.routes.account, 'get', { expand: 'account' })
+      .then(returnValue => {
+        if (returnValue.returnCode !== 0 && typeof returnValue.returnCode !== 'undefined') {
+          return returnValue;
+        }
+        const { data } = returnValue.response;
+        if (!data || !data.Account || !data.Account.Id) {
+          return ErrorHandler.prototype.returnError(11);
         }
+        this.accountId = data.Account.Id;
 
-        return returnError(11, err);
-      });
+        return this;
+      })
+      .catch(({ response }) => ErrorHandler.prototype.parseBadResponse(response));
   }
 
-  getDevices(typeIdParams) {
-    if (!this.securityToken) {
-      return Promise.resolve(returnError(13));
-    } else if (typeIdParams === null) {
-      return Promise.resolve(returnError(15));
+  getDevices(deviceTypeParams) {
+    let promise = Promise.resolve(() => null);
+    if (!this.accountId) {
+      promise = Promise.resolve(this.getAccountInfo());
     }
 
-    const typeIds = Array.isArray(typeIdParams) ? typeIdParams : [typeIdParams];
+    let deviceTypes = [];
+    if (deviceTypeParams) {
+      deviceTypes = Array.isArray(deviceTypeParams) ? deviceTypeParams : [deviceTypeParams];
+    }
 
-    for (let i = 0; i < typeIds.length; i += 1) {
-      const typeId = typeIds[i];
-      if (!constants.allTypeIds.includes(typeId)) {
-        return returnError(15);
+    // TODO: Validate device types when we have a more complete list.
+    Object.values(deviceTypes).forEach(deviceType => {
+      if (!Object.values(constants.allDeviceTypes).includes(deviceType)) {
+        // return ErrorHandler.prototype.returnError(15);
       }
-    }
+    });
```

Comment:
Comment out the for loop so it doesn't run.

Reply:
Removed entirely this per your suggestions.